nuxeo-features/nuxeo-platform-notification/nuxeo-platform-notification-core/src/test/java/org/nuxeo/ecm/platform/ec/notification/TestEmailNotification.java

```diff
@@ -0,0 +1,177 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *  
+ * Contributors:
+ *     Salem Aouana
+ */
+
+package org.nuxeo.ecm.platform.ec.notification;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import javax.inject.Inject;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.NuxeoPrincipal;
+import org.nuxeo.ecm.core.event.Event;
+import org.nuxeo.ecm.core.event.EventService;
+import org.nuxeo.ecm.core.event.impl.DocumentEventContext;
+import org.nuxeo.ecm.platform.notification.api.NotificationManager;
+import org.nuxeo.ecm.platform.test.PlatformFeature;
+import org.nuxeo.mail.SmtpMailServerFeature;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+import org.nuxeo.runtime.test.runner.TransactionalFeature;
+
+import net.htmlparser.jericho.Renderer;
+import net.htmlparser.jericho.Source;
+
+/**
+ * Test the whole process: from subscribing to a given event until the reception of the mail notification on this event.
+ *
+ * @since 11.1
+ */
+@RunWith(FeaturesRunner.class)
+@Features({ PlatformFeature.class, SmtpMailServerFeature.class })
+@Deploy("org.nuxeo.ecm.platform.notification.core")
+@Deploy("org.nuxeo.ecm.platform.notification.api")
+@Deploy("org.nuxeo.ecm.platform.url.api")
+@Deploy("org.nuxeo.ecm.platform.url.core")
+@Deploy("org.nuxeo.ecm.platform.notification.core.tests:OSGI-INF/notification-event-listener-contrib.xml")
+public class TestEmailNotification {
+
+    protected static final String DUMMY_NOTIFICATION_NAME = "DummyNotificationToSendMail";
+
+    protected static final String DUMMY_EVENT_NAME = "dummyNotificationToSendMail";
+
+    protected static final String DOCUMENT_NAME = "anyFile";
+
+    protected static final String ANOTHER_DOCUMENT_NAME = "anyFile2";
+
+    @Inject
+    protected CoreSession session;
+
+    @Inject
+    protected NotificationManager notificationManager;
+
+    @Inject
+    protected EventService eventService;
+
+    @Inject
+    protected TransactionalFeature transactionalFeature;
+
+    @Inject
+    protected SmtpMailServerFeature.MailsResult emailsResult;
+
+    protected DocumentModel domain;
+
+    @Before
+    public void before() {
+        domain = session.createDocumentModel("/", "domain", "Domain");
+        domain = session.createDocument(domain);
+    }
+
+    @Test
+    public void shouldReceiveNotificationMailWhenSubscribeToMainDocument() {
```

Comment:
```suggestion
    public void shouldReceiveNotificationMailWhenSubscribeToDocument() {
```
Main sounds weird, having another  test with "ParentDocument" seems enough to distinguish.